pymc_marketing/mmm/budget_optimizer.py

```diff
@@ -182,43 +273,44 @@ def allocate_budget(
             constraints = {"type": "eq", "fun": lambda x: np.sum(x) - total_budget}
             warnings.warn(
                 "Using default equality constraint: The sum of all budgets should be equal to the total budget.",
+                UserWarning,
                 stacklevel=2,
             )
         elif not isinstance(custom_constraints, dict):
             raise TypeError("`custom_constraints` should be a dictionary.")
         else:
             constraints = custom_constraints
 
-        num_channels = len(self.parameters.keys())
+        num_channels = len(self.parameters["channels"])
         initial_guess = np.ones(num_channels) * total_budget / num_channels
         bounds = [
             (
                 (budget_bounds[channel][0], budget_bounds[channel][1])
                 if channel in budget_bounds
                 else (0, total_budget)
             )
-            for channel in self.parameters
+            for channel in self.parameters["channels"]
         ]
 
         if minimize_kwargs is None:
-            minimize_kwargs = {
-                "method": "SLSQP",
-                "options": {"ftol": 1e-9, "maxiter": 1_000},
-            }
+            minimize_kwargs = self.DEFAULT_MINIMIZE_KWARGS
```

Comment:
Can we consider https://github.com/pymc-labs/pymc-marketing/issues/1193 here?

Reply:
Like it, will apply it.

Reply:
Mypy doesn't like:
```python
        if minimize_kwargs is None:
            minimize_kwargs = {
                **self.DEFAULT_MINIMIZE_KWARGS,
                **(minimize_kwargs or {}),
            }
```

Change to 
```
if minimize_kwargs is None:
            minimize_kwargs = self.DEFAULT_MINIMIZE_KWARGS.copy()
        else:
            minimize_kwargs = {**self.DEFAULT_MINIMIZE_KWARGS, **minimize_kwargs}
```